azure-keyvault-cryptography/src/main/java/com/microsoft/azure/keyvault/cryptography/RsaKey.java

```diff
@@ -59,30 +94,95 @@ public RsaKey(String kid, int keySize, Provider provider) throws NoSuchAlgorithm
         _keyPair  = generator.generateKeyPair();
         _provider = provider;
     }
+    
+    /**
+     * Constructor.
+     * 
+     * Generates a new RsaKey with the given KeyPair. The kid is not set.
+     * The keyPair must be an RSAKey.
+     * @param keyPair
+     */
+    public RsaKey(KeyPair keyPair) {
+    	this(null, keyPair, null);
+    }
 
+    /**
+     * Constructor.
+     * 
+     * Generates a new RsaKey with the given KeyPair.
+     * The keyPair must be an RSAKey.
+     * @param kid
+     * @param keyPair
+     */
     public RsaKey(String kid, KeyPair keyPair) {
     	this(kid, keyPair, null);
     }
 
+    /**
+     * Constructor.
+     * 
+     * Generates a new RsaKey with given KeyPair. The given provider is used for algorithm implementation.
+     * The keyPair must be an RSAKey.
+     * @param kid
+     * @param keyPair
+     * @param provider Java security provider
+     */
     public RsaKey(String kid, KeyPair keyPair, Provider provider) {
 
-        if (Strings.isNullOrWhiteSpace(kid)) {
-            throw new IllegalArgumentException("kid");
-        }
```

Comment:
Was removing the check for kid intentional?  It seems like it should still be set, and the default constructor creates a kid.